src/App.tsx

```diff
@@ -1,39 +1,58 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+type State = {
+  hasClock: boolean,
+  clockName: string
+};
 
-        <span className="Clock__time">
-          {today.toLocaleTimeString()}
-        </span>
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  updateNameId = 0;
+
+  componentDidMount(): void {
+    this.updateNameId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('contextmenu', this.hideClockHandler);
+    document.addEventListener('click', this.showClockHandler);
+  }
+
+  componentWillUnmount(): void {
+    window.clearInterval(this.updateNameId);
+    document.removeEventListener('contextmenu', this.hideClockHandler);
+    document.removeEventListener('click', this.showClockHandler);
+  }
+
+  showClockHandler = () => {
+    this.setState({ hasClock: true });
+  };
+
+  hideClockHandler = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: false });
+  };
+
+  render() {
+    const { hasClock } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+        {hasClock && <Clock clockName={this.state.clockName} />}
```

Comment:
you can also destructure `clockName` on line 49: ` const { hasClock, clockName } = this.state;`